src/consensus/difficulty_adjustment.py

```diff
@@ -27,36 +28,36 @@ def _get_blocks_at_height(
     in the path of the peak.
 
     Args:
-        height_to_hash: dict from sub_block_height to header hash (for path to peak)
         sub_blocks: dict from header hash to SubBlockRecord.
         prev_sb: prev_sb (to start backwards search).
         target_sub_block_height: target sub-block to start
         max_num_sub_blocks: max number of sub-blocks to fetch (although less might be fetched)
 
     """
-    if prev_sb.sub_block_height in height_to_hash and height_to_hash[prev_sb.sub_block_height] == prev_sb.header_hash:
-        # Efficient fetching, since we are fetching ancestor blocks within the heaviest chain
-        return [
-            sub_blocks[height_to_hash[uint32(h)]]
-            for h in range(target_sub_block_height, target_sub_block_height + max_num_sub_blocks)
-            if h in height_to_hash
-        ]
-    # slow fetching, goes back one by one
+    if sub_blocks.contains_sub_height(prev_sb.sub_block_height):
+        header_hash = sub_blocks.sub_height_to_hash(prev_sb.sub_block_height)
+        if header_hash == prev_sb.header_hash:
+            # Efficient fetching, since we are fetching ancestor blocks within the heaviest chain
+            block_list: List[SubBlockRecord] = []
+            for h in range(target_sub_block_height, target_sub_block_height + max_num_sub_blocks):
+                if sub_blocks.contains_sub_height(uint32(h)):
```

Comment:
We should assert that this includes this sub height, as opposed to an if statement. if it is not included, we have big issues here, and the code will produice unexpected results